interfaces: add unconfined access to modem-manager

alfonsosanchezbeato opened the pull request:
Let unconfined apps access modem-manager DBus interface, and also let
the plug access modem-manager in classic.

alfonsosanchezbeato commented:
@jdstrand, thanks for the review. Branch refreshed with changes.

Additionally I had to change basedeclaration to allow connecting mmcli to MM daemon.

jdstrand commented on interfaces/builtin/basedeclaration.go:
```diff
@@ -283,7 +283,6 @@ slots:
         - app
     deny-connection: true
   modem-manager:
-    deny-connection: true
```
I realize you did this to work around https://bugs.launchpad.net/bugs/1640874, but this must be removed since the base declaration was working as intended (ie, it is meant to require a snap declaration for connections to be allowed to modem-manager).

alfonsosanchezbeato commented:
Thanks for the review. I have removed the basedeclaration changes. Additionally, I have changed the policy again for plugs in classic, so it is now equivalent to the one in the ofono interface (that is, I consider it is possible to have a confined client using either the m-m snap or the deb on a classic system).

jdstrand commented on interfaces/builtin/modem_manager_test.go:
```diff
@@ -121,9 +121,8 @@ func (s *ModemManagerInterfaceSuite) TestConnectedPlugSnippetUsesSlotLabelOne(c
+	snippet, err := s.iface.ConnectedPlugSnippet(s.plug, s.slot, interfaces.SecurityAppArmor)
 	c.Assert(err, IsNil)
 	c.Assert(string(snippet), testutil.Contains, "peer=(label=unconfined),")
 }
```
Can you add one more test `TestConnectedPlugSnippedUsesUnconfinedLabelNotOnClassic` that makes sure `"peer=(label=unconfined),"` is not present when `release.OnClassic = false`?

alfonsosanchezbeato commented:
@jdstrand, test added

morphis commented:
@mvo5 @niemeyer Not really sure where those autopkg test failures come from. Are they expected to be stable already? If not can you merge this PR?

mvo5 commented:
:+1:
